narwhal/primary/src/core.rs

```diff
@@ -180,6 +179,51 @@ impl Core {
 
         self.highest_received_round = last_round_number;
         self.highest_processed_round = last_round_number;
+
+        // Get latest header from all peers and process it.
+        let peers: Vec<NetworkPublicKey> = self
+            .committee
+            .others_primaries(&self.name)
+            .into_iter()
+            .map(|(_, _, network_key)| network_key)
+            .collect();
+
+        let network = P2pNetwork::new(self.network.network());
+        let mut header_futures = FuturesUnordered::new();
+        let request = LatestHeaderRequest {};
+        for peer in peers.iter() {
+            let network = network.network();
+            let request = request.clone();
+
+            header_futures.push(async move {
+                let _ = &network;
```

Comment:
Is this needed?

Reply:
Yes, in order for the variable not to be dropped

Reply:
I second this question - I would have expected that the `move` in `async move` already moves these variables into the async block, so they wouldn't be dropped regardless? but I could be wrong bc Rust can be subtle :)